main.py

```diff
@@ -1,29 +1,84 @@
 """
 This is an Add-On for DocumentCloud which wraps the tabula-py library
 https://github.com/chezou/tabula-py
-
-It attempts to convert tables from PDFs to CSVs
+It allows you to export tables from a set of PDFs into CSVs and
+download the resulting CSVs as a zip file.
 """
 
+import os
+import sys
+import time
+import fnmatch
+import zipfile
 import tabula
 from documentcloud.addon import AddOn
+from clouddl import grab
 
 
 class Tabula(AddOn):
-    """A tabula Add-On for DocumentCloud"""
+    """A tabula PDF table extraction Add-On for DocumentCloud"""
 
-    def main(self):
+    def fetch_template(self, url):
+        """Fetch the template from either Dropbox or Google Drive"""
+        os.makedirs(os.path.dirname("./out/"), exist_ok=True)
+        if grab(url, "./out/"):
```

Comment:
I think non-gdrive/dropbox support is important.  how about something like:
```
downloaded = grab(url, "./out/")
if downloaded:
    ... < check the files...> ...
else:
    resp = requests.get(url)
    if resp.status_code == 200:
        with open("./out/template.json", "w") as json_file:
            json_file.write(resp.text)
        return True
````

Even if you use the URL, the file will always be at ./out/template.json.  If the request doesn't return 200 you fall through to the error message.  I am assuming the JSON template is never gonna be too big, so no need to stream it.